win32ss/gdi/ntgdi/dibobj.c

```diff
@@ -132,32 +130,30 @@ CreateDIBPalette(
 
             // FIXME: do we need to handle PALETTEINDEX / PALETTERGB macro?
 
-            /* Loop all color indices in the DIB */
-            for (i = 0; i < cColors; i++)
+            /* Use SEH to verify we can READ all data successfully */
+            _SEH2_TRY
             {
-                /* User SEH to verify READ success */
-                Status = STATUS_SUCCESS;
-                _SEH2_TRY
-                {
-                    rgb = prgb[i];
-                }
-                _SEH2_EXCEPT(EXCEPTION_EXECUTE_HANDLER)
-                {
-                    Status = _SEH2_GetExceptionCode();
-                    /* On Read Failure, put zero in Palette */
-                    PALETTE_vSetRGBColorForIndex(ppal, i, 0);
-                }
-                _SEH2_END
-
-                if(NT_SUCCESS(Status))
+                /* Loop all color indices in the DIB */
+                for (i = 0; i < cColors; i++)
                 {
                     /* Get the color value and translate it to a COLORREF */
+                    RGBQUAD rgb = prgb[i];
                     COLORREF crColor = RGB(rgb.rgbRed, rgb.rgbGreen, rgb.rgbBlue);
 
                     /* Set the RGB value in the palette */
                     PALETTE_vSetRGBColorForIndex(ppal, i, crColor);
                 }
             }
+            _SEH2_EXCEPT(EXCEPTION_EXECUTE_HANDLER)
+            {
+                /* Set all palette entries to zero */
+                for (i = 0; i < cColors; i++)
+                {
+                        PALETTE_vSetRGBColorForIndex(ppal, i, 0);
+                }
+            }
+            _SEH2_END
```

Comment:
```suggestion
            _SEH2_END;
```

Reply:
In this file (dibobj.c) there are four instances of _SEH2_END that do not have the following semi-colon and two that do have the following semi-colon. Can you point me to some authoritative documentation on its syntax?

Reply:
It's the syntax we use everywhere else.

Reply:
Even here https://reactos.org/wiki/PSEH there are examples that are shown both with and without the trailing semicolon.

Reply:
only one over all the other ones.

Reply:
OK. You comment is authoritative enough for me. Fixed. Thanks.

Reply:
There are only very few that use this without `;`:
https://git.reactos.org/?p=reactos.git&a=search&h=HEAD&st=grep&s=_SEH2_END%24&sr=1